config/peerpods/podvm/podvm-builder.sh

```diff
@@ -180,12 +206,27 @@ function delete_podvm_image_gallery() {
     return
   fi
 
-  # Check if force option is passed
+  # Check if peer-pods-cm configmap exists
+  if ! check_peer_pods_cm_exists; then
+    echo "peer-pods-cm configmap does not exist. Skipping image gallery deletion"
+    exit 0
+  fi
+
+  # Get the IMAGE_GALLERY_NAME from the IMAGE_GALLERY_NAME annotation key in peer-pods-cm configmap
+  IMAGE_GALLERY_NAME=$(kubectl get configmap peer-pods-cm -n openshift-sandboxed-containers-operator -o jsonpath='{.metadata.annotations.IMAGE_GALLERY_NAME}')
+
+  # If IMAGE_GALLERY_NAME is not set, then exit
+  if [ -z "${IMAGE_GALLERY_NAME}" ]; then
+    echo "IMAGE_GALLERY_NAME is not set in peer-pods-cm. Skipping image gallery deletion"
+    exit 0
+  fi
+
```

Comment:
Am i getting this right? Image gallery will get deleted only if was set by OSC, based on the annotation we set?
And the image itself will get deleted based on IMAGE_ID value (if not changed) right? however, we do have the image_id as annotation as well, would it make sense to use the same mechanism?

Reply:
The way to check if the image gallery was created by OSC is the tag. The annotation just mentions the image gallery being used. The value in the annotation could be a pre-existing gallery that is entered by the admin user in the azure-podvm-image-cm

Reply:
Sorry, i don't get why the difference in implementation is needed, ie. why galleries are deleted based on the annotations values and images are deleted based on the CM key value (data)

Reply:
Oh forgot to mention that the image is created by packer and the only way to add tag is after creation of the image. But I think as you suggested I can use annotation as the source of truth for OSC created resources and use the same logic for gallery and image.

Reply:
I have changed it to use the gallery value from annotation and avoid using tags completely.

Reply:
> Oh forgot to mention that the image is created by packer and the only way to add tag is after creation of the image. But I think as you suggested I can use annotation as the source of truth for OSC created resources and use the same logic for gallery and image. 

oh i missed that, now it make sense, thanks